google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java

```diff
@@ -259,6 +259,21 @@ public void run() {
   }
 
   /**
+   * Clears out frames and variables
+   * Useful for breakpoint delete operations which result in no selected row
+   */
+  public void clearExecutionStack() {
+    SwingUtilities.invokeLater(new Runnable() {
+      @Override
+      public void run() {
+        if (!getXDebugSession().isStopped()) {
+          getXDebugSession().positionReached(new XSuspendContext() {});
```

Comment:
I agree its a little unclear:

Essentially what is going is that its telling IJ the a position was reached with an empty Context. This has the effect of clearing out the variables and the frames from the UI as well (since we are now in an empty context). 

I was trying to find an alternative mechanism to achieve this but wasn't able to (open to suggestions here). 

When you stop a "normal" debug session in IJ, I believe that it calls the stop() method of the XDebugSession which as the effect of clearing out ALL of our snapshot frames and ending the debug session altogether. In our case, we only want to clear out an individual debug location and display an empty stack _while keeping the debug session alive_ (there's no equivalent of this as far as I can tell in normal IJ debug sessions).